.changeset/tame-beans-pay.md

```diff
@@ -0,0 +1,5 @@
+---
+"@snapwp/query": patch
+---
+
+fix: Add is404 flag for 404ed routes
```

Comment:
Our new spec we're no longer "fixing" anything, but exposing a prop for developers to consume.

```suggestion
dev: Expose `QueryEngine.getTemplateData().is404` for custom HTTP status code handling.
```